routes/invitation.js

```diff
@@ -38,34 +40,42 @@ router.put('/:id/approve', auth, async (req, res) => {
       console.log('new user approval');
     }
 
-    //receive is already a member and this is a friend request
+    //receiver is already a member and this is a friend request
     else {
-      //push receiver into sender's contacts and vice versa
-
-      //copies omit password
-      let receiverCopy = {
-        _id: receiver._id,
-        email: receiver.email,
-        name: receiver.name,
-        primaryLanguage: receiver.primaryLanguage,
-        dateJoined: receiver.dateJoined,
-      };
-
-      let senderCopy = {
-        _id: sender._id,
-        email: sender.email,
-        name: sender.name,
-        primaryLanguage: sender.primaryLanguage,
-        dateJoined: sender.dateJoined,
-      };
-
-      sender.contacts.push(receiverCopy);
-      receiver.contacts.push(senderCopy);
+      //if sender or receiver are already friends
+      const alreadyFriends1 = sender.contacts.find(
+        (friend) => friend.email.toString() === receiver.email,
+      );
+      const alreadyFriends2 = receiver.contacts.find(
+        (friend) => friend.email.toString() === sender.email,
+      );
+
+      if (!alreadyFriends1 && !alreadyFriends2) {
+        const newSenderContacts = [...sender.contacts, receiver.toObject()];
+        const newReceiverContacts = [...receiver.contacts, sender.toObject()];
+
+        sender.contacts = newSenderContacts;
+        receiver.contacts = newReceiverContacts;
+      } else {
+        return res.status(400).json({ msg: 'Users already connected' });
+      }
     }
 
+    //I DONT KNOW IF THIS NEEDS TO BE DONE
```

Comment:
Are we checking when an invitation is created an invitation in the other direction already exists? If not, this is a good idea